spec/upload_to_s3_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'tempfile'
+require_relative './spec_helper'
+
+describe Fastlane::Actions::UploadToS3Action do
+  let(:client) { instance_double(Aws::S3::Client) }
+  let(:test_bucket) { 'a8c-wpmrt-unit-tests-bucket' }
+
+  before do
+    allow(Aws::S3::Client).to receive(:new).and_return(client)
+  end
+
+  # Stub head_object to return a specific content_length
+  def stub_s3_head_request(key, content_length)
+    allow(client).to receive(:head_object)
+      .with(bucket: test_bucket, key: key)
+      .and_return(Aws::S3::Types::HeadObjectOutput.new(content_length: content_length))
+  end
+
+  # Allow us to do `.with` matching against a `File` instance to a particular path in RSpec expectations
+  # Because `File.open(path)` returns a different instance of `File` for the same path on each call)
+  RSpec::Matchers.define :file_instance_of do |path|
+    match { |actual| actual.is_a?(File) && actual.path == path }
+  end
+
+  describe 'happy path' do
+    it 'generates a prefix for the key when using auto_prefix:true' do
+      in_tmp_dir do |tmp_dir|
+        file_path = File.join(tmp_dir, 'input_file_1')
+        File.write(file_path, 'Dummy content')
+        expected_key = 'k5w5OY2yQF55HiBXeP9w+F3/Yg4=/subdir/a8c-key1'
```

Comment:
The `SHA1.base64digest('input_file_1')` seems to lead to a base64 string that contains `/`, which I believe could cause issues (especially if that `/` ends up at the very start or very end of the SHA1 base64), so I think we'll need to fix the implementation (e.g. using `hexdigest` instead) to avoid such cases

Reply:
It'd be _super_ rare that it'd cause a problem, but I agree - better to avoid the risk altogether – I used `hexdigest` for cleaner output – nice catch 🙂